lib/collection/src/collection/query.rs

```diff
@@ -217,7 +217,12 @@ impl Collection {
         .await?;
 
         // update timeout
-        let timeout = timeout.map(|timeout| timeout.saturating_sub(start.elapsed()));
+        let timeout = timeout.map(|timeout| {
+            // timeout is calculated with sufficient precision and avoids flooring
+            (timeout.as_secs_f32() - start.elapsed().as_secs_f32())
+                .ceil()
+                .max(1.0) as usize
```

Comment:
Removing this should be fine.

```suggestion
```

In that case it'll always be at least 1 unless it was 0 already. I think that makes sense.